src/braket/parametric/free_parameter.py

```diff
@@ -99,6 +105,22 @@ def to_dict(self) -> dict:
             "name": self.name,
         }
 
+    @staticmethod
+    def _validate_name(name: str) -> None:
```

Comment:
If we only validate for the symbol creation, can we use a `setter` here and combine L58?

Reply:
Override the setter? That seems like overkill.

Reply:
added a setter